pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -118,6 +121,26 @@ public List<String> listSchemaNames(@Context HttpHeaders headers) {
     return _pinotHelixResourceManager.getSchemaNames(headers.getHeaderString(DATABASE));
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/schemas/info")
+  @Authorize(targetType = TargetType.CLUSTER, action = Actions.Cluster.GET_SCHEMA_INFO)
+  @ApiOperation(value = "List all schemas info with count of field specs", notes = "Lists all schemas with field "
+      + "count details")
+  public String getSchemaInfo(@Context HttpHeaders headers)
+      throws JsonProcessingException {
+    List<SchemaInfo> schemasInfo = new ArrayList<>();
+    TableCache cache = _pinotHelixResourceManager.getTableCache();
+    List<String> schemas = _pinotHelixResourceManager.getSchemaNames(headers.getHeaderString(DATABASE));
+    for (String schemaName : schemas) {
+      Schema schema = cache.getSchema(schemaName);
+      if (schema != null) {
+        schemasInfo.add(new SchemaInfo(schema));
```

Comment:
Consider adding an optional field `uncachedSchemas` for schemas not found in cache. We may show this info to the user in the UI. Something like:
```
{
  "schemasInfo": [
    ...
  ],
  "uncachedSchemas": [
    ...
  ]
}
```

Reply:
Done